openhands/core/config/utils.py

```diff
@@ -159,6 +159,8 @@ def load_from_toml(cfg: AppConfig, toml_file: str = 'config.toml'):
                     logger.openhands_logger.warning(
                         f'Unknown key in {toml_file}: "{key}"'
                     )
+                    cfg.extended.add_dict(key, value)
```

Comment:
If we are doing this, then the logging warning just above shouldn't be a warning. At most, maybe it should inform about adding `key` to extended.

But adding this value as a dict inside "extended" confuses me a bit. Please see follow-up.

Reply:
changing to debug - we could indeed remove it too

Reply:
IMHO we can then create a class method `from_dict` rather than add a dict, in which we just assign the dict to its inner one?